src/main/webapp/3Dscript.js

```diff
@@ -176,11 +213,110 @@ function plotOrientation() {
   orientation.rotation.y = THREE.Math.degToRad(poseTransform.rotate);
 }
 
+let time;
+let lat;
+let long;
+let alt;
+let roll;
+let yaw;
+let pitch;
+
+/**
+* get time in readable units
+* @param {double} time raw time.
+* @return {String} readable time.
+ */
+function timeConverted(time) {
+  const unixTimestamp = time;
+  // Create a new JavaScript Date object based on the timestamp
+  // multiplied by 1000 so that the argument is in milliseconds, not seconds.
+  const date = new Date(unixTimestamp * 1000);
+  // Hours part from the timestamp
+  const hours = date.getHours();
+  // Minutes part from the timestamp
+  const minutes = '0' + date.getMinutes();
+  // Seconds part from the timestamp
+  const seconds = '0' + date.getSeconds();
+
+  // Will display time in 10:30:23 format
+  const formattedTime = hours + ':' +
+   minutes.substr(-2) + ':' + seconds.substr(-2);
+  return formattedTime;
+}
+
+/**
+* Does matrix multiplication on specific index of the direction matrix.
+* @param {object} typeOfMatrix matrix to multiply by.
+*/
+function changeMatrix(typeOfMatrix) {
+  direction.getMatrixAt(increment, instanceMatrix );
+  matrix.multiplyMatrices( instanceMatrix, typeOfMatrix );
+  direction.setMatrixAt( increment, matrix );
+  direction.instanceMatrix.needsUpdate = true;
+}
+
+/**
+* Hides beginning of trajectory based on user selected point.
+ */
+function hideOrientationStart() {
```

Comment:
Is it possible to pass more of the values here explicitly? It's hard to follow with all of the global interactions. (It might also let you use a single function instead of having both `hideOrientationStart` and `hideOrientationEnd`.)

Reply:
next commit doesnt pass new values, but condenses to 1 hideorientation function